Property_map/include/CGAL/Dynamic_property_map.h

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2007  GeometryFactory (France).  All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org); you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public License as
+// published by the Free Software Foundation; either version 3 of the License,
+// or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+//
+//
+// Author(s)     : Andreas Fabri
+
+#ifndef CGAL_DYNAMIC_PROPERTY_MAP_H
+#define CGAL_DYNAMIC_PROPERTY_MAP_H
+
+#include <boost/graph/graph_traits.hpp>
+#include <CGAL/boost/graph/properties.h>
+#include <CGAL/property_map.h>
+#include <boost/shared_ptr.hpp>
+#include <boost/unordered_map.hpp>
+
+namespace CGAL {
+
+namespace internal {
+
+template <typename K, typename V>
+struct Dynamic_property_map {
+
+  typedef K key_type;
+  typedef V value_type;
+  typedef const value_type& reference;
+  typedef boost::read_write_property_map_tag  category;
+
+  Dynamic_property_map(const V& default_value = V())
+    : map_(new Map()), default_value_(default_value)
+  {}
+
+  void clear()
+  {
+    map_ = boost::shared_ptr<Map>(0);
+  }
```

Comment:
There is [compilation issue][1], related to that function.

  [1]: https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.11-Ic-65/Polygon_mesh_processing/TestReport_lrineau_Fedora-strict-ansi.gz

Anyway, I do not see what it does: the current `shared_ptr` drop its reference. But in `remove_property` the `pm` object is passed by copy, and that means that in the end the `Map` will still survive. Why not do the following?
```c++
void clear() {
  if(map_)  map_->clear();
}
```